### PR TITLE
Update secondary skill handling. Fix issue 2307

### PR DESCRIPTION
**Pull request by dydzio0614**
Also making secondary skill icon display behavior same as in H3 during popup message.

### CODE REVIEW

File: lib/mapObjects/CGPandoraBox.cpp [hunk 1]
```diff
@@ -68,32 +68,51 @@ void CGPandoraBox::giveContentsUpToExp(const CGHeroInstance *h) const
 
 	if(gainedExp || changesPrimSkill || abilities.size())
 	{
+		std::vector<SecondarySkill> unpossessedAbilities;
```
Comment (DjWarmonger) [inline lib/mapObjects/CGPandoraBox.cpp]: I suggest std::pair<Secondary Skill, int>

File: lib/mapObjects/CGPandoraBox.cpp [hunk 2]
```diff
@@ -68,32 +68,51 @@ void CGPandoraBox::giveContentsUpToExp(const CGHeroInstance *h) const
...
+		std::vector<int> unpossessedAbilityLevels;
+		int abilitiesRequiringSlot = 0;
+
+		//filter out unnecessary secondary skills
+		for (int i = 0; i < abilities.size(); i++)
```
Comment (DjWarmonger) [inline lib/mapObjects/CGPandoraBox.cpp]: use auto

File: lib/mapObjects/CGPandoraBox.cpp [hunk 3]
```diff
@@ -68,32 +68,51 @@ void CGPandoraBox::giveContentsUpToExp(const CGHeroInstance *h) const
...
+			
 		for(int i=0; i<primskills.size(); i++)
 			if(primskills[i])
 				iw.components.push_back(Component(Component::PRIM_SKILL,i,primskills[i],0));
 
-		for(int i=0; i<abilities.size(); i++)
-			iw.components.push_back(Component(Component::SEC_SKILL,abilities[i],abilityLevels[i],0));
-
-		cb->showInfoDialog(&iw);
+		for(int i=0; i<unpossessedAbilities.size(); i++)
```
Comment (DjWarmonger) [inline lib/mapObjects/CGPandoraBox.cpp]: same - auto will give shorter code

**Comment by ArseniyShestakov** [comment]
@DjWarmonger can you please clarify how exactly you expect to use auto in that case?

He actually use "i" to access multiple different vectors and this looks like reasonable approach to me if we don't want to change object serialization.

**Comment by ArseniyShestakov** [comment]
Ok I got it. It's actually possible to use auto some of loops if he won't duplicate object properites structure there and just use std::pair. Agree on it.

**Comment by DjWarmonger** [comment]
Thank you very much :)

**Pull request merged**